### ISSUE TITLE
About pos_position_ids and neg_position_ids

### ISSUE THREAD

**Issue by lmk0724** [bytedance/HLLM#24]
In HLLM/code/REC/data/dataset/trainset.py, there is this code: 
`pos_position_ids.extend((torch.array(len(ids) + self.item_emb_token_n) + (self.max_ text_length - len(ids))).tolist())`
may I ask why add an offset like `self.max_text_length - len(ids)`? I think because the ids may perform a padding operation, but I didn't find a padding operation for the ids.

**Comment by Geek-Yang-Zhe** [comment]
I have the same problem.It seems that ids is not padded, which will cause the pos_input_ids dimension to be inconsistent. Stacking in the customize_rmpad_collate function should result in an error
'ids = self.tokenizer.encode(text_str)
ids = ids[:self.max_text_length]'
'pos_input_ids.extend(ids + [0] * self.item_emb_token_n)'
' if isinstance(elem, torch.Tensor):
out = None
if torch.utils.data.get_worker_info() is not None:
# If we're in a background process, concatenate directly into a
# shared memory tensor to avoid an extra copy
numel = sum(x.numel() for x in batch)
storage = elem.storage()._new_shared(numel)
out = elem.new(storage)
return torch.stack(batch, 0, out = out)'

**Comment by Geek-Yang-Zhe** [comment]
I have the same problem. It seems that the ids are not padded, which will cause the pos_input_ids dimension to be inconsistent. Stacking in the customize_rmpad_collate function should result in an error.
In trainset.py
`ids = self.tokenizer.encode(text_str) 
ids = ids[:self.max_text_length]`
`pos_input_ids.extend(ids + [0] * self.item_emb_token_n)`
In customize_rmpad_collate:
` if isinstance(elem, torch.Tensor):
...
        return torch.stack(batch, 0, out=out)`

**Comment by ssyzeChen** [comment]
Hi~
We use rmpad operation to faster training with flash attention, you could check codes in https://github.com/bytedance/HLLM/blob/main/code/REC/data/dataset/collate_fn.py#L131
